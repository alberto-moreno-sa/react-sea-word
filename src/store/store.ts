import configs from 'configs';
import {
  createStore,
  applyMiddleware,
  combineReducers,
  StoreEnhancer,
  Store,
  Reducer,
  AnyAction,
} from 'redux';
import { createWrapper, Context, HYDRATE } from 'next-redux-wrapper';
import thunkMiddleware from 'redux-thunk';
import { seaWordReducer, seaWordInitialState } from './seaWord';
import { createSelectorHook } from 'react-redux';

interface SeaWordStore extends Store {}

const combinedReducer = combineReducers({
  seaWord: seaWordReducer,
});

export type RootState = ReturnType<typeof combinedReducer>;
export const useSelector = createSelectorHook<RootState>();
export const RESET_STORE = 'RESET_STORE';

export const InitialState: RootState = {
  seaWord: seaWordInitialState,
};

export interface PersistedState extends RootState {
  persistVersion: string;
}

export const reducer = (state = InitialState, action: AnyAction): RootState => {
  if (action.type === HYDRATE) {
    const nextState: RootState = {
      ...state,
      ...action.payload,
    };

    return nextState;
  }

  if (action.type === RESET_STORE) {
    return { ...InitialState };
  }
  return combinedReducer(state, action);
};

const bindMiddleware = (middleware): StoreEnhancer => {
  if (configs.env === 'local') {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    const { composeWithDevTools } = require('redux-devtools-extension');
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    const { logger } = require('redux-logger');
    middleware.push(logger);
    return composeWithDevTools(applyMiddleware(...middleware));
  }

  return applyMiddleware(...middleware);
};

export let store: SeaWordStore = null;
const makeConfiguredStore = (r: Reducer<RootState>): Store =>
  createStore(r, InitialState, bindMiddleware([thunkMiddleware]));

export const makeStore = (context: Context): Store => {
  store = makeConfiguredStore(reducer);

  return store;
};

// export an assembled wrapper
export const wrapper = createWrapper<Store<RootState>>(makeStore, {
  debug: configs.env === 'local',
});
