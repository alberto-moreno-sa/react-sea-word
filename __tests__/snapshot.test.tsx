import React from 'react';
import { render } from 'testUtils';
import Index from 'pages/index';
import { InitialState } from 'store/store';

it('renders Home details unchanged', () => {
  const { container } = render(<Index />, {
    initialState: InitialState,
  });

  expect(container).toMatchSnapshot();
});
