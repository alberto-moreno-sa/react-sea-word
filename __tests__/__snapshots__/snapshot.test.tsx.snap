// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home details unchanged 1`] = `
<div>
  <div
    data-testid="sea-word-home"
  >
    <nav
      class="flex flex-wrap items-center justify-between py-2.5 px-3 mb-3 bg-light-blue-500"
      data-testid="nav-bar"
    >
      <div
        class="container max-w-7xl px-4 mx-auto flex flex-wrap items-center justify-between"
      >
        <div
          class="w-full flex justify-between lg:w-auto lg:static lg:block lg:justify-start"
        >
          <a
            data-testid="nav-bar-link"
            href="/"
          >
            <span
              class="text-sm font-bold leading-relaxed inline-block mr-4 py-2 whitespace-no-wrap uppercase text-white"
            >
              SeaWord
            </span>
          </a>
          <button
            class="cursor-pointer text-xl leading-none py-1 px-1.5 rounded-full border border-solid border-transparent bg-transparent block lg:hidden outline-none focus:outline-none hover:bg-white hover:bg-opacity-20 transition-all duration-300"
            data-testid="test-nav-toggler"
          >
            <span
              class="block relative w-6 h-px rounded-sm bg-white"
            />
            <span
              class="block relative w-6 h-px rounded-sm bg-white mt-1"
            />
            <span
              class="block relative w-6 h-px rounded-sm bg-white mt-1"
            />
          </button>
        </div>
        <div
          class="lg:flex flex-grow items-center hidden"
          data-testid="test-menu"
        >
          <ul
            class="flex lg:items-center flex-col lg:flex-row list-none ml-auto"
          >
            <a
              class="px-5 py-4 flex gap-1 items-center text-xs uppercase font-medium leading text-white rounded-lg"
              data-testid="test-nav-menu-link"
              href="https://www.linkedin.com/in/albert-moreno/"
              rel="noreferrer"
              target="_blank"
            >
              <i
                class="fab fa-brands fa-linkedin-in text-xl leading-none"
                data-testid="icon"
              />
               My Profile
            </a>
            <a
              class="px-5 py-4 flex gap-1 items-center text-xs uppercase font-medium leading text-white rounded-lg"
              data-testid="test-nav-menu-link"
              href="https://github.com/alberto-moreno-sa/sea-word-challenge"
              rel="noreferrer"
              target="_blank"
            >
              <i
                class="fab fa-github text-xl leading-none"
                data-testid="icon"
              />
               Github
            </a>
          </ul>
        </div>
      </div>
    </nav>
    <main
      data-testid="main-content"
    >
      <section
        class="relative py-16 bg-gray-100"
      >
        <div
          class="container max-w-7xl px-4 mx-auto"
        >
          <div>
            hello
          </div>
        </div>
      </section>
    </main>
    <footer
      class="relative bg-gray-100 pt-8 pb-6"
      data-tesid="footer"
    >
      <div
        class="container max-w-7xl mx-auto px-4"
      >
        <div
          class="flex flex-wrap text-center lg:text-left pt-6"
        >
          <div
            class="w-full lg:w-6/12 px-4"
          >
            <h1
              class="text-2xl font-serif font-bold leading-normal mt-0 mb-2"
            >
              Sea Word
            </h1>
            <p
              class="text-lg font-light leading-relaxed mt-6 mb-4"
              style="margin-top: 0px;"
            >
              Small Sea word where you cnat create island.
            </p>
            <div
              class="flex gap-2 mt-6 md:justify-start md:mb-0 mb-8 justify-center"
            >
              <a
                class="grid place-items-center bg-white text-gray-900 shadow-md font-normal h-10 w-10 items-center justify-center align-center rounded-full outline-none focus:outline-none"
                href="https://github.com/alberto-moreno-sa/sea-word-challenge"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="fab fa-github text-xl"
                  data-testid="icon"
                />
              </a>
              <a
                class="grid place-items-center bg-white text-gray-900 shadow-md font-normal h-10 w-10 items-center justify-center align-center rounded-full outline-none focus:outline-none"
                href="https://www.linkedin.com/in/albert-moreno/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="fab fa-brands fa-linkedin-in text-xl "
                  data-testid="icon"
                />
              </a>
            </div>
          </div>
        </div>
        <hr
          class="my-6 border-gray-300"
        />
        <div
          class="flex flex-wrap items-center md:justify-between justify-center"
        >
          <div
            class="w-full md:w-4/12 px-4 mx-auto text-center"
          >
            <div
              class="text-sm text-gray-700 font-medium py-1"
            >
              Copyright © 
              2022
               by
               
              <a
                class="text-gray-700 hover:text-gray-900 transition-all"
                href="https://github.com/alberto-moreno-sa"
              >
                Alberto Moreno
              </a>
              .
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
